straxen/plugins/peak_processing.py

```diff
@@ -324,3 +325,153 @@ def find_n_competing(peaks, windows, fraction):
             n_tot[i] = n_left[i] + np.sum(areas[i + 1:right_i] > threshold)
 
         return n_left, n_tot
+
+
+@export
+class VetoPeakTags(IntEnum):
+    """Identifies by which detector peak was tagged.
+    """
+    # Peaks are not inside any veto interval
+    NO_VETO = 0
+    # Peaks are inside a veto interval issued by:
+    NEUTRON_VETO = 1
+    MUON_VETO = 2
+    BOTH = 3
+
+
+@export
+class PeakVetoTagging(strax.Plugin):
+    """
+    Plugin which tags S1 peaks according to  muon and neutron-vetos.
+    Tagging S2s is does not make sense as they occur with a delay.
+    However, we compute for both S1/S2 the time delay to the closest veto
+    region.
+
+        * untagged: 0
+        * neutron-veto: 1
+        * muon-veto: 2
+        * both vetos: 3
+    """
+    __version__ = '0.0.1'
+    depends_on = ('peak_basics', 'veto_regions_nv', 'veto_regions_mv')
+    provides = ('peak_veto_tags')
+    save_when = strax.SaveWhen.TARGET
+
+    dtype = strax.time_fields + [
+        ('veto_tag', np.int8,
+         'Veto tag for S1 peaks. unatagged: 0, nveto: 1, mveto: 2, both: 3'),
+        ('time_to_closest_veto', np.int64)
+    ]
+
+    def get_time_difference(self, peaks, veto_regions_nv, veto_regions_mv):
+        """
+        Computes time differences to closest nv/mv veto signal.
+
+        It might be that neutron-veto and muon-veto signals overlap
+        Hence we compute first the individual time differences to the
+        corresponding vetos and keep afterwards the smallest ones.
+        """
+        dt_nv = get_time_to_closest_veto(peaks, veto_regions_nv)
+        dt_mv = get_time_to_closest_veto(peaks, veto_regions_mv)
+
+        dts = np.transpose([dt_nv, dt_mv])
+        ind_axis1 = np.argmin(np.abs(dts), axis=1)
+        return self._get_smallest_value(dts, ind_axis1)
+
+    @staticmethod
+    @numba.njit(cache=True, nogil=True)
+    def _get_smallest_value(time_differences, index):
+        res = np.zeros(len(time_differences), np.int64)
+        for res_ind, (ind, dt) in enumerate(zip(index, time_differences)):
+            res[res_ind] = dt[ind]
+        return res
```

Comment:
is it true that we cannot do something like this:
```python
>>> import numpy as np; from pprint import pprint
>>> a = np.random.randint(0, 10, 10)
>>> b = np.random.randint(0, 10, 10)
>>> pprint([a,b])
[array([0, 7, 7, 4, 6, 2, 5, 9, 7, 5]), array([9, 5, 3, 8, 4, 0, 3, 5, 4, 6])]
>>> np.min([a,b], axis=0)
array([0, 5, 3, 4, 4, 0, 3, 5, 4, 5])
```
because dt_mv or dt_nv may be negative? Probably makes sense but I guess people we need to keep an eye out that people will assume it's always positive (certainly worth adding a description to the 'time_to_closest_veto'-field)

Reply:
Yes, the time difference can be negative as it contains information if the closest veto interval is before or after the corresponding event/peak. This is an important information as neutron-veto has to tag peaks/events before the vetoing-event in the neutron-veto while for the muon-veto it is the other way around. So this information may be needed for cross-checks and optimization. I will extend the description.